stratum/hal/lib/common/yang_parse_tree_paths.cc

```diff
@@ -3377,7 +3436,10 @@ void YangParseTreePaths::AddSubtreeInterfaceFromTrunk(
 void YangParseTreePaths::AddSubtreeInterfaceFromSingleton(
     const SingletonPort& singleton, const NodeConfigParams& node_config,
     YangParseTree* tree) {
-  const std::string& name = singleton.name();
+  const std::string& name =
+      !singleton.name().empty()
+          ? singleton.name()
+          : absl::StrFormat("%d/%d", singleton.port(), singleton.slot());
```

Comment:
You also need to include the channel number since we can have multiple interfaces per channel on same physical port+slot

Reply:
So the complete order would be slot/port/channel?